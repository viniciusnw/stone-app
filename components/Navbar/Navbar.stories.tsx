import React from "react";
import { View } from "react-native";
import styled from "styled-components/native";
import { Navbar } from "@Components";

const Wrapper = styled(View)`
  height: 100%;
  justify-content: flex-end;
`;

type Nav = {
  icon: string;
  label: string;
  onPress: () => any;
  primary: boolean;
};

const nav: Nav[] = [
  {
    icon: "wallet",
    label: "Cateira",
    onPress: () => true,
    primary: false,
  },
  {
    icon: "plus",
    label: "Carrinho",
    onPress: () => true,
    primary: false,
  },
  {
    icon: "logout",
    label: "Logout",
    onPress: () => true,
    primary: false,
  },
];

export const Default = {};

export default {
  title: "Navbar",
  component: () => (
    <Wrapper>
      <Navbar items={nav} />
    </Wrapper>
  ),
};
