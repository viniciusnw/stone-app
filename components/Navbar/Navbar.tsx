import React from "react";
import styled from "styled-components/native";
import { View, Text, TouchableOpacity, TouchableHighlight } from "react-native";
import Icon from "@Components/Icon";

const StyledContainer = styled(View)`
  background-color: #ddd;
  padding: 0 20px;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  height: ${({ theme }) => theme.height.navbar};
`;

const StyledItem = styled(TouchableOpacity)`
  align-items: center;
`;

const StyledLabel = styled(Text)`
  font-weight: 700;
  font-size: 11px;
  color: ${({ theme }) => theme.color.gray};
  text-transform: uppercase;
`;

const StyledPrimary = styled(TouchableHighlight)`
  width: 62px;
  height: 62px;
  border-radius: 100px;
  background-color: ${({ theme }) => theme.color.black};
  justify-content: center;
  align-items: center;
  margin-top: -40px;
  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.4);
`;

type Props = {
  items: Array<{
    icon: string;
    label: string;
    onPress: () => null;
    primary: boolean;
  }>;
};

const Navbar: React.FC<Props> = ({ items, ...props }: any) => {
  return (
    <StyledContainer {...props}>
      {items.map(({ icon, primary, label, ...props }: any, idx: number) =>
        primary ? (
          <StyledPrimary key={idx} {...props}>
            <Icon icon={icon} size={32} color="#fff" />
          </StyledPrimary>
        ) : (
          <StyledItem key={idx} {...props}>
            <>
              <Icon icon={icon} size={28} />
              <StyledLabel>{label}</StyledLabel>
            </>
          </StyledItem>
        )
      )}
    </StyledContainer>
  );
};

export default Navbar;
