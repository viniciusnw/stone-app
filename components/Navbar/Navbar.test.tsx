import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Navbar from "./Navbar";

jest.useFakeTimers();

type Nav = {
  icon: string;
  label: string;
  onPress: () => any;
  primary: boolean;
};

const nav: Nav[] = [
  {
    icon: "wallet",
    label: "Cateira",
    onPress: () => true,
    primary: false,
  },
  {
    icon: "plus",
    label: "Carrinho",
    onPress: () => true,
    primary: false,
  },
  {
    icon: "logout",
    label: "Logout",
    onPress: () => true,
    primary: false,
  },
];

test("Navbar", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Navbar items={nav} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
