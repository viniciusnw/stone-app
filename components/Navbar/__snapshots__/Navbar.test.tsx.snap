// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "flexDirection": "row",
      "height": 62,
      "justifyContent": "space-around",
      "paddingBottom": 0,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 0,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        {
          "color": "#666666",
          "fontSize": 11,
          "fontWeight": "700",
          "textTransform": "uppercase",
        }
      }
    >
      Cateira
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        {
          "color": "#666666",
          "fontSize": 11,
          "fontWeight": "700",
          "textTransform": "uppercase",
        }
      }
    >
      Carrinho
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        {
          "color": "#666666",
          "fontSize": 11,
          "fontWeight": "700",
          "textTransform": "uppercase",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
